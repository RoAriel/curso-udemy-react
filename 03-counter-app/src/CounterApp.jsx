import { useState } from "react"
import PropTypes from "prop-types"

export const CounterApp = ({ value }) => {

    const [count, setCount] = useState(value)

    const handleAdd = () => { setCount(count + 1) }
    const handleSubstract = () => { setCount(count - 1) }
    const handleRest = () => { setCount(value) }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{count}</h2>

            <button aria-label='btn-add' onClick={handleAdd}> +1 </button>
            <button aria-label='btn-subst' onClick={handleSubstract}> -1 </button>
            <button aria-label='btn-reset' onClick={handleRest}> Reset </button>
        </>
    )
}

CounterApp.prototype = {
    value: PropTypes.number.isRequired,
}