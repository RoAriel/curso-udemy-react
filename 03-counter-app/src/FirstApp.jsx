import PropTypes from "prop-types"

export const FirstApp = ({ title = 'No title', subTitle = 'No sub', name }) => {

    return (
        <>
            <h1 data-testid='test-title'>{title}</h1>
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.prototype = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}
