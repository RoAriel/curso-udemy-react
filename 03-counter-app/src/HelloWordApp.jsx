export const HelloWordApp = () => {
  return (
    <h1>
      Hello World App
    </h1>
  )
}

