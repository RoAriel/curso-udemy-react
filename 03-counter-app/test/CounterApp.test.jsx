import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Test en <CounterApp />', () => {

    const initialValue = 100

    test('should match con snapshot', () => {
      
        const {container} = render(<CounterApp value={initialValue}/>)
        expect(container).toMatchSnapshot();
    })
    
    test('should mostrar valor inicial 100', () => {
      
        render(<CounterApp value={initialValue} />)
        expect(screen.getByRole('heading', {level:2}).innerHTML).toContain(initialValue.toString())
    })

    test('should incrementar con el +1', () => {
      
        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByText('+1'))
        expect(screen.getByRole('heading', {level:2}).innerHTML).toContain('101')


    })

    test('should decrementar con el -1', () => {
      
        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByRole('heading', {level:2}).innerHTML).toContain('99')

    })

    test('should reset el contador', () => {
      
        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByText('-1'))
        fireEvent.click(screen.getByText('-1'))

        //forma de encontrar el boton
        fireEvent.click(screen.getByRole('button',{name: 'btn-reset'}))
        expect(screen.getByRole('heading', {level:2}).innerHTML).toContain('100')


    })
    
    
  
})
