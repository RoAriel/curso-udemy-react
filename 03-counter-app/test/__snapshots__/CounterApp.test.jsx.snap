// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <CounterApp /> should match con snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button
    aria-label="btn-add"
  >
     +1 
  </button>
  <button
    aria-label="btn-subst"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
